scripts/rebalance.ts

```diff
@@ -93,9 +103,48 @@ async function main() {
             if (!(await isStrategyEligibleForNSR(strategyAddress))) {
               continue;
             }
+
+            const strategy = IRebalancingV2Strategy__factory.connect(strategyAddress, signer)
             const strategyName = await IStrategyV2__factory.connect(strategyAddress, ethers.provider).strategySpecificName();
             console.log('Processing strategy', strategyName, strategyAddress);
 
+            const getBlockTimeStamp = async () => (await ethers.provider.getBlock(await ethers.provider.getBlockNumber())).timestamp
```

Comment:
for some reason this kind of logic is buggy (sometimes, idk the reason)
use Misc.getBlockTsFromChain